### PR TITLE
feat: extended ScannedProject to include a new field in meta for gradle projects naming convetion

### PR DESCRIPTION
**Pull request by sfat**
- [ ] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
related to https://github.com/snyk/snyk-gradle-plugin/issues/93

### CODE REVIEW

File: legacy/plugin.ts [hunk 1]
```diff
@@ -89,7 +89,8 @@ export interface PluginMetadata {
 
   // Per-plugin custom metadata
   meta?: {
-    allSubProjectNames?: string[];
+    allSubProjectNames?: string[],
+    gradleProjectName?: Array<Map<string, string>>;
```
Comment (lili2311) [inline legacy/plugin.ts]: Looks like this can just be `string[]`? What is the data example this will hold?

Comment (sfat) [reply legacy/plugin.ts]: format will be:
```
{
  "packageName": "oldPackageName",
  "newPackageName:" "newPackageName" 
}
```

The same format will be for multi projects as well, part of `ScannedProject`

File: legacy/plugin.ts [hunk 2]
```diff
@@ -100,6 +100,9 @@ export interface PluginMetadata {
 export interface SinglePackageResult {
   plugin: PluginMetadata;
   package: DepTree;
+  meta?: {
+    gradleProjectName?: any,
```
Comment (lili2311) [inline legacy/plugin.ts]: why not `string`? it is always a string right?

Comment (sfat) [reply legacy/plugin.ts]: good idea.
will change it to `string`

**Comment by sfat** [comment]
squashed changes.

**Pull request merged**